File: majorityElement.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        List<Integer> result = new ArrayList<>();
        HashMap<Integer,Integer> pairs = new HashMap<Integer, Integer>();
        
        for(int i = 0; i<nums.length; i++){
            if((pairs.get(nums[i])==null)){
                pairs.put(nums[i],1);
            }else{
                int value=pairs.get(nums[i]);
                pairs.put(nums[i],value+1);        
            }
        }
        for (Integer i : pairs.keySet()) {
            if(pairs.get(i)>(nums.length/3))result.add(i);
        }   
        return result;
    }
}

File: go/majorityElement.go
func majorityElement(nums []int) int {
	if len(nums) == 1 {
		return nums[0]
	}
	occurencies := make(map[int]int)
	condition := len(nums) / 2

	for i := 0; i < len(nums); i++ {
		val, ok := occurencies[nums[i]]
		if !ok {
			occurencies[nums[i]] = 1
		} else if val == condition {
			return nums[i]
		} else {
			occurencies[nums[i]] = val + 1
		}
	}

	return 42
}